cli/src/commands/scan.rs

```diff
@@ -433,20 +443,44 @@ fn print_rules_as_text(
     // `the `by_ref` method cannot be invoked on a trait object`
     #[allow(clippy::while_let_on_iterator)]
     while let Some(matching_rule) = rules.next() {
-        let line = if print_namespace {
+        let name = if print_namespace {
             format!(
-                "{}:{} {}",
+                "{}:{}",
                 matching_rule.namespace().paint(Cyan).bold(),
-                matching_rule.identifier().paint(Cyan).bold(),
-                file_path.display(),
+                matching_rule.identifier().paint(Cyan).bold()
             )
         } else {
-            format!(
-                "{} {}",
-                matching_rule.identifier().paint(Cyan).bold(),
-                file_path.display()
-            )
+            format!("{}", matching_rule.identifier().paint(Cyan).bold())
+        };
+
+        let meta = if print_meta {
+            let mut meta_str: String = String::from("");
```

Comment:
Instead of allocating a new string for the metadata, and then using `format!` again for concatenating the name and the metadata (which causes yet another allocation), I'll simply rename `name` back to `line`, make it mutable, and append the metadata information to it.